packages/block-editor/src/components/global-styles/get-block-css-selector.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { get, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { scopeSelector } from './utils';
+
+export function getBlockCSSSelector(
+	blockType,
+	target = 'root',
+	options = {}
+) {
+	if ( ! target ) {
+		return null;
+	}
+
+	const { fallback = false } = options;
+	const { name, selectors, supports } = blockType;
+
+	const hasSelectors = ! isEmpty( selectors );
+	const path = Array.isArray( target ) ? target.join( '.' ) : target;
+
+	// Duotone ( no fallback selectors for Duotone ).
+	if ( path === 'filter.duotone' ) {
+		// If selectors API in use, only use its value or null.
+		if ( hasSelectors ) {
+			return get( selectors, path, null );
+		}
```

Comment:
I didn't comment on this in the previous PR because I felt like it could be included in the duotone follow-up, but I do think it would be better for duotone to use fallback selectors to match everything else.

I want to avoid as much duotone-specific logic as possible—that was the point of making `supports.__experimentalDuotone` experimental—eventually we could remove the duotone specifics.

Reply:
> I want to avoid as much duotone-specific logic as possible—that was the point of making supports.__experimentalDuotone experimental—eventually we could remove the duotone specifics.

A quick search of the WP plugin directory showed some hits for `__experimentalDuotone` out in the wild so I'm not sure it's a simple removal, we'd need to follow an official deprecation process right? In the same way, we can't just remove the `__experimentalSelector` support.

We'll need some form of specific duotone code here as its old location under `color.__experimentalDuotone` doesn't follow the general `feature.__experimentalSelector` pattern.

>  I do think it would be better for duotone to use fallback selectors to match everything else

So to confirm, it's ok for a duotone selector, if requested, to fallback to `filter.root` or the block's root selector?

Combining this with the need to update duotone block support to not scope its selectors, it might be better in a separate follow-up to this PR.  

What do you think about landing this now (assuming style variation updates get greenlit) to fix the glaring issues before 15.5RC and create a separate PR to address this immediately after?